<?xml version="1.0"?>
<launch>
  <arg name="dataRoot" default="$(env HOME)" />
  <arg name="locationsRoot" default="$(arg dataRoot)/pbd_location_saved/" />
  <arg name="map_file" default="$(find rws_pr2_navigation)/config/map.yaml" />

  <include file="$(find pr2_2dnav)/launch/pr2_2dnav.launch" />
  <node name="robot_pose_publisher" pkg="robot_pose_publisher" type="robot_pose_publisher" />

  <node name="rws_pr2_navigation" pkg="rws_pr2_navigation" type="main.py" output="screen">
    <param name="locationsRoot" value="$(arg locationsRoot)" />
  </node>

  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" respawn="true" />

  <!-- This is the stuff required for web-based GUI -->
  <node pkg="tf2_web_republisher" type="tf2_web_republisher" name="tf2_web_republisher" output="screen" />
</launch>
